PukingPredator/Assets/Scripts/SceneTransition.cs

```diff
@@ -0,0 +1,75 @@
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+using System.Collections;
+
+public class SceneTransition : SingletonMonobehaviour<SceneTransition>
+{
+    [SerializeField]
+    private Image fadePrefab;
+
+    private Image fadeImage;
+
+    private string nextSceneName;
+
+    public void FadeToScene(string sceneName)
+    {
+        Canvas canvas = FindObjectOfType<Canvas>();
+        if (canvas)
+        {
+            fadeImage = Instantiate(fadePrefab, canvas.transform);
+
+            nextSceneName = sceneName;
+            StartCoroutine(FadeIn());
+        } else
+        {
+            SceneManager.LoadScene(sceneName);
+        }
+    }
+
+    private IEnumerator FadeOut()
+    {
+        Canvas canvas = FindObjectOfType<Canvas>();
+        fadeImage = Instantiate(fadePrefab, canvas.transform);
+
+        float timeElapsed = 0f;
+        float fadeDuration = 1f;
+
+        while (timeElapsed < fadeDuration)
+        {
+            timeElapsed += Time.deltaTime;
+            
+            // Lerps alpha to transparent
+            fadeImage.color = new Color(0f, 0f, 0f, Mathf.Lerp(1f, 0f, timeElapsed / fadeDuration));
+
+            if (timeElapsed < fadeDuration + fadeDuration / 2) 
+            yield return null;
+        }
```

Comment:
it needs to delete the fadeImage after its done. it currently is causing a menu bug where you cant interact with the pause menu at all because its in the way